packages/e2e-tests/utils/SelfReloadingJSONRpcProvider.js

```diff
@@ -0,0 +1,46 @@
+const { JsonRpcProvider } = require("near-api-js/lib/providers");
+const { BN } = require("bn.js");
+const { parseNearAmount } = require("near-api-js/lib/utils/format");
+
+const { createAccountWithHelper } = require("../services/contractHelper");
+const E2eTestAccount = require("./E2eTestAccount");
+const { generateTestAccountId, getWorkerAccountRegex } = require("./helpers");
+const nearApiJsConnection = require("./connectionSingleton");
+const { getTestAccountSeedPhrase } = require("./helpers");
+
+class SelfReloadingJSONRpcProvider extends JsonRpcProvider {
```

Comment:
Per offline conversation, let's make sure we cache the pending promise for `reloadAccount()` and chain on it if we are performing multiple parallel requests, chain on the existing pending reload request